### PR TITLE
Vet Clinic Database Project  Part 2

### PR DESCRIPTION
**Pull request by Profsain**
I completed this part of the project by carrying out the following:

- Added new column name species to animals table
- Inserted 7 new records to animals table
- Updated species column
- Created database transactions to update, delete and implement transaction control
- Wrote database queries to answer the following:
      How many animals are there?
      How many animals have never tried to escape?
      What is the average weight of animals?
      Who escapes the most, neutered or not neutered animals?
      What is the minimum and maximum weight of each type of animal?
      What is the average number of escape attempts per animal type of those born between 1990 and 2000?

Attached is the screenshot of my results

Species Update operation

crt1

Update weight operation
Delete Rollback operation
crt2
crt3

Queries Operation
LST1
LST2

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -25,4 +25,22 @@ SELECT * FROM animals WHERE name != 'Gabumon';
 with the weights that equals precisely 10.4kg or 17.3kg)*/
 SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
 
-
+-- Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
+UPDATE animals SET species='digimon' WHERE name LIKE '%mon';
+-- Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
+UPDATE animals SET species='pokemon' WHERE species = '';
```
Comment (DeliceLydia) [inline queries.sql:31]: - [ ] I suggest that after updating these queries you must commit you can do:
```
 SELECT species from animals; -- verify that change was made
   COMMIT;
   SELECT species from animals; -- verify that change persists after commit

```

Comment (DeliceLydia) [reply queries.sql:31]: - [ ] Please work on `UPDATE` the animal's table by setting the species column to unspecified, then roll back the change:
```SQL
BEGIN; -- start a transaction

   UPDATE animals
   SET species = 'unspecified'; -- make change

   SELECT species from animals; -- verify that change was made
   ROLLBACK;

   SELECT species from animals; -- verify that change was undone
```

Comment (DeliceLydia) [reply queries.sql:31]: - [ ] Please delete all records in the animal's table, then roll back the transaction:
```sql
BEGIN;

   DELETE FROM animals WHERE ;
   SELECT COUNT(*) FROM ANIMALS;

   ROLLBACK;
   SELECT COUNT(*) FROM ANIMALS;
```

Comment (DeliceLydia) [reply queries.sql:31]: - [ ] Please inside a transaction work on:
- `DELETE` all animals born after Jan 1st, 2022
- Create a savepoint for the transaction
- `UPDATE` all animals' weight to be their weight multiplied by -1.
- Rollback to the savepoint
- `UPDATE` all animals' negative weights to be their weight multiplied by -1.
- Commit transaction
you can do the following:
```sql
 BEGIN;

    DELETE FROM animals WHERE  date_of_birth >= '2022/01/01' ;
    SAVEPOINT SP1;

    UPDATE animals SET weight_kg = weight_kg * -1; 
    ROLLBACK TO SP1;
    UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0; 

    COMMIT;
```

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @Profsain, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

**Review by mukhammadrizooff — APPROVED** [review]
Hi @Profsain, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- Good implementations ✔️ 
- Gave correct values ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**